src/map/packets/message_basic.cpp

```diff
@@ -41,6 +42,22 @@ CMessageBasicPacket::CMessageBasicPacket(CBaseEntity* PSender, CBaseEntity* PTar
     ref<uint16>(0x18) = messageID;
 }
 
+CMessageBasicPacket::CMessageBasicPacket(CBaseEntity* PSender, CBaseEntity* PTarget, int32 param, int32 value, MsgStd messageID)
```

Comment:
There's no need to copy the full implementation of this constructor again.

```cpp
#include <iostream>

enum SomeEnum : int
{
    SomeThing  = 1,
    OtherThing = 2,
};

class SomePacket
{
public:
    SomePacket(int key)
    {
        // Do something with int key
        std::cout << "Constructing SomePacket with: " << key << "\n";
    }

    SomePacket(SomeEnum asEnum)
    : SomePacket(static_cast<int>(asEnum)) // 
    {
    }

};

int main()
{
    auto packet = SomePacket(SomeEnum::SomeThing);
    return 0;
}
```

See https://en.cppreference.com/w/cpp/language/constructor#Delegating_constructor

Reply:
🙏🏼 Just what I needed, thanks!